### PR TITLE
Docs: Add warning in tiered storage

### PR DESCRIPTION
**Pull request by Fokko**

### CODE REVIEW

File: docs/spark-procedures.md [hunk 1]
```diff
@@ -514,6 +514,10 @@ will then treat these files as if they are part of the set of files  owned by Ic
 `expire_snapshot` calls will be able to physically delete the added files. This method should not be used if 
 `migrate` or `snapshot` are possible.
 
+{{< hint warning >}}
+Keep in mind the `add_files` procedure will fetch the Parquet metadat from each file being added just once. If you're using tiered storage, (such as [Amazon S3 Intelligent-Tiering storage class](https://aws.amazon.com/s3/storage-classes/intelligent-tiering/)), the file will be retrieved from the archive, and will be on a higher tier for a set period of time.
```
Comment (singhpk234) [inline docs/spark-procedures.md:518]:
```suggestion
Keep in mind the `add_files` procedure will fetch the Parquet metadata from each file being added just once. If you're using tiered storage, (such as [Amazon S3 Intelligent-Tiering storage class](https://aws.amazon.com/s3/storage-classes/intelligent-tiering/)), the underlying, file will be retrieved from the archive, and will remain on a higher tier for a set period of time.
```

**Review by singhpk234 — COMMENTED** [review]
Thanks for adding this @Fokko !

File: docs/spark-procedures.md (hunk 1 above)
Comment (singhpk234) [reply docs/spark-procedures.md:518]: [minor / optional] we can also add a small explanation of why this metadata will be fetched, (for ex: getting stats for manifest and manifest_list files)

Comment (Fokko) [reply docs/spark-procedures.md:518]: Great addition @singhpk234 thanks. Since it is a warning, I don't want it to be too lengthy.

**Review by singhpk234 — APPROVED** [review]
LGTM, Thanks @Fokko !

**Comment by Fokko** [comment]
Thanks @nastra and @singhpk234 for the review!

**Pull request merged**